addons/mail/models/mail_activity.py

```diff
@@ -328,6 +328,8 @@ def create(self, values):
             self.env['bus.bus'].sendone(
                 (self._cr.dbname, 'res.partner', activity.user_id.partner_id.id),
                 {'type': 'activity_updated', 'activity_created': True})
+        if activity.res_model and activity.res_id and activity.activity_type_id.decoration_type in ['warning', 'danger']:
+            self.env[activity.res_model].browse(activity.res_id)._set_activity_exception_type()
```

Comment:
We actually have no certitude that self.env[activity.res_model].browse(activity.res_id) is a mail_activity_mixin i think.